gen/installer/aws.py

```diff
@@ -532,50 +533,51 @@ def get_button(template_name, s3_url):
 
 
 def do_create(tag, build_name, reproducible_artifact_path, commit, variant_arguments, all_bootstraps):
-    # Generate the single-master and multi-master templates.
-
-    for bootstrap_variant, variant_base_args in variant_arguments.items():
-        # Setup base arguments
-        args = deepcopy(variant_base_args)
-        args['exhibitor_address'] = '{ "Fn::GetAtt" : [ "InternalMasterLoadBalancer", "DNSName" ] }'
-        args['s3_bucket'] = '{ "Ref" : "ExhibitorS3Bucket" }'
-        args['s3_prefix'] = '{ "Ref" : "AWS::StackName" }'
-        args['exhibitor_storage_backend'] = 'aws_s3'
-        args['master_role'] = '{ "Ref" : "MasterRole" }'
-        args['agent_role'] = '{ "Ref" : "SlaveRole" }'
-        args['region_to_ami_mapping'] = gen_ami_mapping({"stable"})
-        args['nat_ami_mapping'] = gen_ami_mapping({"natami"})
-
-        variant_prefix = pkgpanda.util.variant_prefix(bootstrap_variant)
-
-        def make(gen_args, filename):
-            gen_out = gen_templates(gen_args)
-            yield from _as_artifact_and_pkg(variant_prefix, filename, gen_out)
-
-        # Single master templates
-        single_args = deepcopy(args)
-        single_args['num_masters'] = "1"
-        yield from make(single_args, 'single-master.cloudformation.json')
-
-        # Multi master templates
-        multi_args = deepcopy(args)
-        multi_args['num_masters'] = "3"
-        yield from make(multi_args, 'multi-master.cloudformation.json')
-
-        # Advanced templates
-        for os_type in ['coreos', 'el7']:
-            yield from gen_advanced_template(
-                variant_base_args,
-                variant_prefix,
-                reproducible_artifact_path,
-                os_type)
-
-    # Button page linking to the basic templates.
-    button_page = gen_buttons(build_name, reproducible_artifact_path, tag, commit, variant_arguments)
-    yield {
-        'channel_path': 'aws.html',
-        'local_content': button_page,
-        'content_type': 'text/html; charset=utf-8'}
-
-    # This renders the infra template only, which has no difference between CE and EE
-    yield from gen_supporting_template()
+    with logger.scope("Creating AWS installer"):
```

Comment:
Why does this need to be inside each do_create function? Why can't we do it at the site that calls all of these that has the module name as well as the do_create function? Can pretty easily add the name to use as an extra piece of data it pulls out of the module:

https://github.com/dcos/dcos/blob/master/release/__init__.py#L401

Saves a lot of indenting everywhere, making code easier / simpler to read and harder to get wrong (or accidentally forget). Forgetting becomes a hard error.